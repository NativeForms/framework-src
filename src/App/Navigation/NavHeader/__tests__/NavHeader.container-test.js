import React from 'react';
import { Button } from 'native-base';
import sinon from 'sinon';
import { shallow, mount } from 'enzyme';

// app
import NavHeaderContainer from '../NavHeader.container';

const props = {
  navigation: { navigate: sinon.spy() },
  scene: {},
  getScreenDetails: sinon.spy(() => ({ options: {} }))
};

describe('components <NavHeaderContainer />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavHeaderContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  const consoleError = global.console.error;
  beforeEach(() => {
    global.console.error = jest.fn();
  });

  it('should toggleMenu on button press', () => {
    const wrapper = mount(<NavHeaderContainer {...props} />);
    wrapper.find(Button).props().onPress();
    expect(props.navigation.navigate.calledOnce).toBeTruthy();
    expect(props.getScreenDetails.calledOnce).toBeTruthy();
  });

  afterEach(() => {
    global.console.error = consoleError;
  });
});
