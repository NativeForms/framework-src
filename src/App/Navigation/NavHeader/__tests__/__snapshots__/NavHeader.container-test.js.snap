// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <NavHeaderContainer /> should render correctly 1`] = `
<NavHeaderComponent
  getScreenDetails={[Function]}
  navigation={
    Object {
      "navigate": [Function],
    }
  }
  onToggleMenu={[Function]}
  scene={Object {}}
/>
`;
